backon/src/lib.rs

```diff
@@ -33,14 +33,42 @@
 //!
 //! Retry in BackON requires an implementation for sleeping. BackON will accept a [`Sleeper`] to pause for a specified duration.
 //!
-//! BackON employs the following default sleep implementations:
+//! ## Default `Sleeper`
 //!
-//! - `tokio-sleep`: Utilizes [`TokioSleeper`] within a Tokio context in non-wasm32 environments.
-//! - `gloo-timers-sleep`: Utilizes [`GlooTimersSleep`] to pause in wasm32 environments.
+//! Currently, BackON has 2 built-in `Sleeper` implementations for different environments,
+//! they are gated under their own features, which are enabled by default:
 //!
-//! Users CAN provide a custom implementation if they prefer not to use the default options.
+//! |      `Sleeper`      | feature          | Environment |
+//! |---------------------|------------------|-------------|
+//! | [`TokioSleeper`]    | tokio-sleep      | non-wasm32  |
+//! | [`GlooTimersSleep`] | gloo-timers-sleep|   wasm32    |
 //!
-//! If neither feature is enabled nor a custom implementation is provided, BackON will fallback to an empty sleeper. This will cause a panic in the `debug` profile and do nothing in the `release` profile.
+//! ## Custom `Sleeper`
+//!
+//! If you do not want to use the built-in `Sleeper`, you CAN provide a custom
+//! implementation, here is an example that implements a `Sleeper` with `monoio::time::sleep`:
+//!
+//! ```rust,ignore
+//! use std::time::Duration;
+//! use backon::Sleeper;
+//! use monoio::time::sleep;
+//! use monoio::time::Sleep;
+//!
+//! /// Sleeper implemented using `monoio::time::sleep()`.
+//! struct MonoioSleeper;
+//!
+//! impl Sleeper for MonoioSleeper {
+//!     type Sleep = Sleep;
+//!
+//!     fn sleep(&self, dur: Duration) -> Self::Sleep {
+//!         sleep(dur)
+//!     }
+//! }
+//! ```
+//!
+//! ## The empty `Sleeper`
+//!
+//! If neither feature is enabled nor a custom implementation is provided, BackON will fallback to the empty sleeper. This will cause a panic in the `debug` profile and do nothing in the `release` profile.
```

Comment:
This is out-of-date now. We will raise a build error now.

Reply:
Updated